packages/expect/src/utils.ts

```diff
@@ -299,7 +302,7 @@ export const subsetEquality = (
       }
       const result =
         object != null &&
-        hasOwnProperty(object, key) &&
+        hasPropertyInObject(object, key) &&
```

Comment:
**Why not just replace `hasOwnProperty`?**

It seems like this is being used elsewhere (i.e. `getObjectSubset`) and adding the `in` check caused unit tests to break. I can't find what is referencing `getObjectSubset` but since it's publicly exported, we leaned towards not touching it 😅

Reply:
It's probably important to note that this thing will probably cause something like this to pass:

```
expect({}).toMatchObject({ toString: expect.any(Function) });
```

But, I don't think that's necessarily a bad thing 🤔

Reply:
Can we exclude methods from `Object.prototype`?

Reply:
> Can we exclude methods from Object.prototype?

I think so. We could probably do something like `key in obj && !(key in Object.prototype)`. @SimenB what do you think?

On the other hand, I'm not sure what the benefit is of removing the `Object.prototype` keys. **Technically** someone could set a value to `toString` or `valueOf`, and I think we'd want to be unbiased and include it 🤔. I don't feel strongly either way, so your call 😅

Reply:
Good point. Can we recurse the prototype chain up until we reach `Object`?

Reply:
> Can we recurse up the prototype chain up until we reach Object and use hasOwnProperty on the way?

Yes! Great idea. I went ahead and added two new commits for this:

1. Update the `hasPropertyInObject` implementation to recurse up the prototype chain (stopping at `Object.prototype`)
2. Replace references to the previous `hasOwnProperty` function, since this is now redundant and a bit complex in comparison.

@SimenB could you please take another look? :smile: